File: typescript-client-sdk/src/sdk/models/operations/getonewebhook.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetOneWebhookQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=webhookId" })
  webhookId?: string;
}


export class GetOneWebhookResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=webhook" })
  webhook?: shared.WebhookMetadata;
}


export class GetOneWebhookRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetOneWebhookQueryParams;
}


export class GetOneWebhookResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetOneWebhookResponseBody;
}

File: go-client-sdk/pkg/models/operations/getonewebhook.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetOneWebhookQueryParams struct {
	AccessToken *string `queryParam:"style=form,explode=true,name=accessToken"`
	WebhookID   *string `queryParam:"style=form,explode=true,name=webhookId"`
}

type GetOneWebhookResponseBody struct {
	Webhook *shared.WebhookMetadata `json:"webhook,omitempty"`
}

type GetOneWebhookRequest struct {
	QueryParams GetOneWebhookQueryParams
}

type GetOneWebhookResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetOneWebhookResponseBody
}
